src/processor/processor.module.ts

```diff
@@ -43,14 +46,24 @@ import { ProviderWebhookService } from './provider-webhook.service';
     BullModule.registerQueue({
       name: 'graphUpdateQueue',
       defaultJobOptions: {
-        attempts: 1,
+        attempts: 3,
         backoff: {
           type: 'exponential',
         },
         removeOnComplete: true,
-        removeOnFail: true,
+        removeOnFail: false,
```

Comment:
Be careful with this: our jobIds are not unique (the original idea was that we should only have a single job to update a particular user graph at any given time. But then the problem is that it turns out jobId needs to be unique even across failed jobs, I think. 

This might need some scenario testing to make sure we can still enqueue a job for the same user+graph+provider even if a previous job failed.

We might need to add an additional uniqueness component to the job ID... but then we might need to do some extra concurrency checking in the queue processor.

Reply:
I think this mostly retention? jobs that will fail more than 3 times will just goto failed jobs, I have run some tests
Not sure if you mean something else

But this is about retention right? and not clearaway failed jobs

Reply:
What I mean is, if we have a failed job retained with ID "provider:msa:recurse", will it prevent enqueuing a subsequent job for the same ID? I think it will; meaning we have to either:
* Have additional uniqueness in the job ID
* Use delete-then-add pattern for enqueuing jobs
* Use 'update-or-add' pattern for enqueuing jobs

Reply:
Yes it will, we can add additional uniqueness, wondering we can use attemptsMade,

Reply:
So each attemp becomes a separate job, which should be fine considering the state will get refreshed on next update

Reply:
I added a dead letter mechanism, any failed jobs we can mark with dead letter and make sure we dont re-queue them upon failure again

Reply:
Do we want to keep *all* failed jobs indefinitely? This parameter can also be set to a number, specifying the number of failed jobs to keep;
[Queue#add](https://github.com/OptimalBits/bull/blob/master/REFERENCE.md#queueadd)

Reply:
I think we should keep failed job, ideally the dead letter should get them to completion but if still failure occurs i believe its worth keep them failed

Reply:
Do you have number in mind?